testsuite/smoke-test/src/consensus/consensusdb_recovery.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    smoke_test_environment::new_local_swarm_with_aptos,
+    test_utils::{assert_balance, create_and_fund_account, transfer_coins},
+};
+use consensus::CONSENSUS_DB_NAME;
+use forge::{HealthCheckError, NodeExt, Swarm};
+use std::{
+    fs,
+    time::{Duration, Instant},
+};
+
+#[tokio::test]
+async fn test_consensusdb_recovery() {
```

Comment:
Can you add a test that ensures epoch change? May be just shorten the epoch length ane ensure we wait for some time for an epoch change?

Reply:
epoch change won't test this path, state sync would catch up to the new epoch and start with round manager

Reply:
Correct me if I am missing something but its possible that the recovery manager thread is doing state sync, while the epoch manager receives an epoch chaging proof, thus terminating the recreating a new recovery manager. Do you think there is a way to test that code path? I agree it should be a corner case though.

Reply:
it can not terminating an existing state sync, the close request needs to wait until the state sync finishes